### PR TITLE
utils: directory for utilities separation

### PR DESCRIPTION
**Pull request by niewim19**
JIRA: PP-95

## Description

 - ekf development applcications (stored in ekf/tests/ directory) are moved to separate utils directory
 - `altlog` utility heavily modified to `senslog` utility which provides selections of;
   - total sampling time
   - sampling interval
   - devices to sample from
   - corrections enable/disable

## Motivation and Context

For development purposes it is nice to have ability for flexible sensor logging. Currently it involved directly changing code of `devekf` or `sensortest` and it produced unreliable logs (each implementation was different).

It will be decided if similar application will be developed for EKF or EKF will be embeded into this app in the future.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: utils/senslog.c [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+/* Parsing arguments Returns 0 on success, -1 when parsing failed */
+static int senslog_parseArgs(int argc, char **argv)
+{
+	int c;
+	const char *devs = NULL;
+
+	senslog_common.step = 0;
+	senslog_common.diffMax = 0;
+	senslog_common.init = false;
+	senslog_common.raw = false;
```
Comment (HBuczynski) [inline utils/senslog.c:113]: These variables equal `0` by default.

File: utils/senslog.c [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+	sensor_event_t accelEvt, gyroEvt, magEvt, baroEvt;
+	char buf[128];
+	time_t t0, now, diff;
+	int parse, cnt;
+
+	parse = senslog_parseArgs(argc, argv);
+	if (!senslog_common.init) {
+		/* Check if parsing failed, or only help was printed */
+		return (parse == 0) ? EXIT_SUCCESS : EXIT_FAILURE;
+	}
```
Comment (HBuczynski) [inline utils/senslog.c:171]: The `senslog_common.init` has the same value as `parse`.

File: utils/senslog.c [hunk 3]
```diff
@@ -0,0 +1,239 @@
+/*
+ * Phoenix-Pilot
+ *
+ * extended kalman filter
```
Comment (HBuczynski) [inline utils/senslog.c:4]: You can change that it is a part of utility.

File: utils/Makefile [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#
+# Makefile for utility applications in Phoenix-Pilot project
+#
+# This source code and Makefile is designed for compilation from _user directory of phoenix-rtos-project
```
Comment (HBuczynski) [inline utils/Makefile:4]: Ups, it is not a `_user` directory.

**Review by HBuczynski — CHANGES_REQUESTED** [review]

File: utils/senslog.c [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+		if (senslog_common.flags.baro) {
+			sensc_baroGet(&baroEvt);
+		}
+		/* Above ifs to be expanded with new sensors */
+
+		memset(buf, 0, sizeof(buf));
+		cnt = 0;
+
+		/* printing time as a fake float value */
+		cnt += sprintf(&buf[cnt], "%lli.%06lli ", diff / 1000000, (diff - 1000000 * (diff / 1000000)));
```
[the remark this answers is not in the document]
Comment (niewim19) [reply utils/senslog.c]: Thanks, I missed that. Added logic for this error check (whole line is skipped if such error occurs)

File: utils/senslog.c [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+				return -1;
+		}
+	}
+
+	if (senslog_common.step == 0 || senslog_common.diffMax == 0 || senslog_parseDevices(devs) < 0) {
+		fprintf(stderr, "Invalid arguments!\n");
+		return -1;
+	}
+
+	senslog_common.init = true;
```
[the remark this answers is not in the document]
Comment (niewim19) [reply utils/senslog.c]: As a matter of fact - no. I have removed it from the code and used value returned by `senslog_parseArgs` to indicate if application shall be stopped with or without an error.

File: utils/senslog.c (hunk 1 above)
Comment (niewim19) [reply utils/senslog.c]: Yes, you`re right, it is not necessary. 

But in scope of this function imho it is more clear when it asserts correct initialization of parameters (`if` condition at the end of this function). If it is not against our coder style I would like to leave it as is.

File: utils/senslog.c [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+	/* Main logging loop */
+	while (diff < senslog_common.diffMax) {
+		gettime(&now, NULL);
+		diff = now - t0;
+
+		if (senslog_common.flags.accel || senslog_common.flags.gyro || senslog_common.flags.mag) {
+			sensc_imuGet(&accelEvt, &gyroEvt, &magEvt);
+		}
+		if (senslog_common.flags.baro) {
```
Comment (HBuczynski) [inline utils/senslog.c:200]: There are separate conditions:
```suggestion
		}
		
		if (senslog_common.flags.baro) {
```

File: utils/senslog.c [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+	if (argc < 2) {
+		senslog_help(argv[0]);
+		return -1;
+	}
+
+	while ((c = getopt(argc, argv, "ht:s:d:r")) != -1) {
+		switch (c) {
+			case 'h':
+				senslog_help(argv[0]);
+				return +1; /* only help message printed, returning +1 */
```
Comment (HBuczynski) [inline utils/senslog.c:127]:
```suggestion
				return 1; /* only help message printed, returning 1 */
```

File: utils/senslog.c [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+
+	return -1;
+}
+
+
+/* 
+* Parsing arguments. Returns:
+* 0 if all variables were successfully initialized, and application is good to go.
+* -1 if an error occurred during parsing
+* +1 if only help was printed and application shall exit.
```
Comment (HBuczynski) [inline utils/senslog.c:107]:
```suggestion
* 1 if only help was printed and application shall exit.
```

File: utils/senslog.c [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+}
+
+/* Parse devices log selection options */
+static int senslog_parseDevices(const char *devs)
+{
+	bool err = false;
+
+	if (devs == NULL) {
+		return -1;
+	}
```
Comment (HBuczynski) [inline utils/senslog.c:60]: Applying this change, you will need to check only an `err` value in `95` line.
```suggestion
	if (devs == NULL || strlen(devs) == 0) {
		return -1;
	}
```

Comment (niewim19) [reply utils/senslog.c:60]: Good idea! I have applied this check as separate if clause just below the highlighted one to be more secure (to not rely with condition output on conditions order)

**Review by HBuczynski — APPROVED** [review]

**Pull request merged**